server/controllers/rateController.js

```diff
@@ -0,0 +1,79 @@
+import Sequelize from 'sequelize';
+import Model from '../models';
+
+const { Article, Rate } = Model;
+
+/**
+ *
+ *
+ * @class RateArticle
+ */
+class RateController {
+/**
+  * @description -This method rates an article
+  * @param {object} req - The request payload sent from the router
+  * @param {object} res - The response payload sent back from the controller
+  * @returns {object} - status, message and articles details
+  */
+  static rateArticle(req, res) {
+    if (Number.isNaN(Number(req.body.rating)) !== true) {
+      Article.findOne({
+        where: { slug: req.params.slug }
+      }).then((article) => {
+        if (article !== null) {
+          Rate.findOne({
+            where: { articleId: article.dataValues.id, userId: req.user.id },
+          })
+            .then((rate) => {
+              if (rate) {
+                return rate.update({
+                  userId: req.user.id,
+                  articleId: article.dataValues.id,
+                  rating: req.body.rating
+                });
+              }
+              return Rate.create({
+                userId: req.user.id,
+                articleId: article.dataValues.id,
+                rating: req.body.rating
```

Comment:
I put this in a single line but the team's eslint settings with "--fix" auto corrects it, placing them in separate lines.